packages/validator-query/lib/AmpValidatorQuery.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class AmpValidatorQuery {
+  constructor(rules) {
+    this.initRules_(rules);
+  }
+
+  getTagsForFormat(format, transformed) {
+    format = format.toLowerCase();
+    return this.tags
+        .filter(
+            (tag) =>
+              tag.htmlFormat.includes(format.toUpperCase()) &&
+          this.checkEntityFormat_(tag, format) &&
+          this.checkEntityTransformed_(tag, transformed)
+        )
+        .map((tag) => {
+          tag = Object.assign({}, tag);
+          tag.attrs = tag.attrs.filter(
+              (attr) =>
+                this.checkEntityFormat_(attr, format) &&
+            this.checkEntityTransformed_(attr, transformed)
+          );
+          return tag;
+        });
+  }
+
+  getExtensionsForFormat(format) {
```

Comment:
Add JSDoc

Reply:
I'd expect this method to return an array. Why return an object instead?

Reply:
-   There shouldn't ever be more than one extension with the same name and...
-   ...I assumed the query would usually be "give me information about amp-bind", so an object seems more appropriate. Also, it's cheap to transform an object back into an array with `Object.keys` and/or `Object.values`.

Reply:
How about turning this into two different APIs? 

```
const ampBindRule = validationRules.getExtension('amp-bind');

const extensions = validationRules.getExtensions() <- returns an array containing all extensions
const mailExtensions = validationRules.getExtensions({format: 'AMP4EMAIL'}) <- with optional format parameter
```

Reply:
I removed `getExtensions` and instead exposed it as an `extensions` property and added `getExtension` which follows your suggestion.